### PR TITLE
[DOCS] Clarify logger-package relationship

### LINKED ISSUE #64669: [DOCS] Document all possible logging options
We should document the available loggers and their uses.  

Examples:
logger.org.elasticsearch.common.NodeConnectionsService -- Supports: Info, Debug, Trace -- Purpose:  Useful for debugging if a node is having issues connecting to other nodes.

logger.org.elasticsearch.xpack.security.authc -- Supports: Info, Debug, Trace -- Purpose: Useful for troubleshooting authentication issues especially with LDAP auth. 

logger.org.elasticsearch.transport 

We have a ton of logger.org.* options, and it would help to have the various supported options and an explanation available.

### PR DESCRIPTION
**Pull request by jrodewig**
Updates the logging level docs to better clarify the relationship
between loggers and their Java packages in lieu of an exhaustive
list.

Closes #64669

### Preview
https://elasticsearch_65169.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/logging.html#configuring-logging-levels

### CODE REVIEW

File: docs/reference/setup/logging-config.asciidoc [hunk 1]
```diff
@@ -139,67 +139,51 @@ documentation].
...
--------------------------------
-PUT /_cluster/settings
-{
-  "transient": {
-    "<name of logging hierarchy>": "<level>"
-  }
-}
--------------------------------
-// NOTCONSOLE
+Each Java package in the {es-repo}[{es} source code] has a related logger. For
```
Comment (jrodewig) [inline docs/reference/setup/logging-config.asciidoc:142]: > Each Java package ...

I'm not 100% sure if this is the case. If not, maybe we can say `many` or `most`?

File: docs/reference/setup/logging-config.asciidoc [hunk 2]
```diff
@@ -139,67 +139,51 @@ documentation].
 [[configuring-logging-levels]]
 === Configuring logging levels
 
-There are four ways to configuring logging levels, each having situations in which they are appropriate to use.
-
-1. Via the command-line: `-E <name of logging hierarchy>=<level>` (e.g.,
-   `-E logger.org.elasticsearch.transport=trace`). This is most appropriate when
-   you are temporarily debugging a problem on a single node (for example, a
-   problem with startup, or during development).
```
Comment (jrodewig) [inline docs/reference/setup/logging-config.asciidoc:147]: I intentionally dropped the command line example here. This seems to be primarily for development and wouldn't be used much in production. Let me know if that's wrong or if we need to include it for another reason.

**Review by pgomulka — APPROVED** [review]
it looks really good.
but I think we need to expand on the `cluster settings API` way of chaning log levels. There is some confusion on how to use it now.
I have an issue to clarify this first https://github.com/elastic/elasticsearch/issues/65208
we can merge this PR and I could work on this later once we decide on the #65208

**Comment by jrodewig** [comment]
Thanks @pgomulka! I agree. I'll go ahead and merge this in. We can update these docs with a follow-up PR once #65208 is decided.

**Pull request merged**